AsyncDisplayKit/Details/ASChangeSetDataController.mm

```diff
@@ -98,6 +98,10 @@ - (void)endUpdatesAnimated:(BOOL)animated completion:(void (^)(BOOL))completion
     for (_ASHierarchyItemChange *change in [_changeSet itemChangesOfType:_ASHierarchyChangeTypeInsert]) {
       [super insertRowsAtIndexPaths:change.indexPaths withAnimationOptions:change.animationOptions];
     }
+    
+    for (_ASHierarchyItemChange *change in [_changeSet itemChangesOfType:_ASHierarchyChangeTypeMove]) {
+      [super moveRowAtIndexPath:change.fromIndexPath toIndexPath:change.toIndexPath withAnimationOptions:change.animationOptions];
+    }
```

Comment:
Here's some sample code demonstrating it using UICollectionView. Ignore the project name 🙂
[MoveExample.zip](https://github.com/facebook/AsyncDisplayKit/files/688555/MoveExample.zip)